src/axes/axis.ts

```diff
@@ -129,6 +139,14 @@ export class Axis<D> extends Component {
     this._tickLabelContainer = this.content().append("g")
                                            .classed(Axis.TICK_LABEL_CLASS + "-container", true);
     this._baseline = this.content().append("line").classed("baseline", true);
+    this._annotationContainer = this.content().append("g")
+                                              .classed("annotation-container", true);
+    this._annotationContainer.append("g").classed("annotation-line-container", true);
+    this._annotationContainer.append("g").classed("annotation-circle-container", true);
+    this._annotationContainer.append("g").classed("annotation-rect-container", true);
```

Comment:
I tried this, and could not get it to work as intended.  The problem is I want the texts and the rectangles to appear on top of all of the lines.  This means that all of the lines must appear first, then the rectangles, then the texts.

I could theoretically have everything reside a single container and then insert each piece in a specified order; I just thought the container idea would be more organized.